addons/stock_account/models/stock.py

```diff
@@ -435,10 +435,10 @@ def product_price_update_before_done(self, forced_qty=None):
         # adapt standard price on incomming moves if the product cost_method is 'average'
         std_price_update = {}
         for move in self.filtered(lambda move: move._is_in() and move.product_id.cost_method == 'average'):
-            product_tot_qty_available = move.product_id.qty_available + tmpl_dict[move.product_id.id]
+            product_tot_qty_available = move.product_id.with_context(owner_id=False).qty_available + tmpl_dict[move.product_id.id]
             rounding = move.product_id.uom_id.rounding
 
-            qty_done = move.product_uom._compute_quantity(move.quantity_done, move.product_id.uom_id)
+            qty_done = move.product_uom._compute_quantity(move.with_context(exclude_owner=True).quantity_done, move.product_id.uom_id)
```

Comment:
You should probably add api.depends_context('exclude_owner') on _quantity_done_compute

Reply:
@amoyaux It seems that `api.depends_context` has been introduced in 13.0
This fix is only for 12.0.

Reply:
The issue is not present in 13.0.
So it will only be forward-ported to saas-12.3